### PR TITLE
Add Facebook authentication

### LINKED ISSUE #3: Feature: Facebook and Google Auth
We need to add social auth. I'm also open to other auths as well, but Facebook and Google were requested. The requirements are:

1. 1 url, for social auth taking in the name as a param
2. A library to handle controller logic
3. The route must return the JWT. We already use JWT in the normal auth

### PR DESCRIPTION
**Pull request by andrewmarklloyd**
Added facebook auth, closes #3. The google auth has been added as a new issue

### CODE REVIEW

File: server/routes/auth.route.js [hunk 1]
```diff
@@ -19,4 +50,7 @@ router.route('/register')
 router.route('/random-number')
   .get(expressJwt({ secret: config.jwtSecret }), authCtrl.getRandomNumber);
 
+router.route('/facebook/callback')
```
Comment (andrewmarklloyd) [inline server/routes/auth.route.js]: @TheHollidayInn you originally said the endpoint should take the name of the social network as a param. Do you advise against explicitly defining each route like this?

Comment (TheHollidayInn) [reply server/routes/auth.route.js]: Nah, I don't feel strongly wither way. Not unless there is a ton of duplicate business logic, which there ins't.

**Comment by andrewmarklloyd** [comment]
@sw-yx I don't think we need to use `express-session` since the API will only return a `jwt` token and doesn't maintain a session

**Comment by sw-yx** [comment]
ok

**Pull request merged**